community-tutorials/install-web-server-mariadb-debian-php/01-en.md

```diff
@@ -0,0 +1,382 @@
+
+# Requirements
+
+You need a Debian server with an internet connection. The server should be reachable from the internet via a static IP on port 80 and 443.<br>
+You also need root permissions on your server.
+
+# Step 1 - Apache2
+
+## Step 1.1 - Installing snapd and Certbot
+
+First we need to install snapd and Certbot to obtain SSL certificates:
+
+```sh
+apt update && apt upgrade -y && apt autoremove -y
+apt install curl sudo wget apache2-utils nano git snapd cron zip unzip tar screen openssl -y
+snap install core; snap refresh core
+sudo snap install --classic certbot
+sudo ln -s /snap/bin/certbot /usr/bin/certbot
+{ crontab -l 2>/dev/null; echo "$(( $RANDOM % 60 )) $(( $RANDOM % 3 + 3 )) * * * sudo certbot renew --dry-run" ; } | crontab -
+```
+
+## Step 1.2 - Installing Apache2
+
+Now we are going to install Apache2 and configure it:
+
+```sh
+apt install apache2 libapache2-mod-php brotli -y
+a2enmod rewrite headers env dir mime proxy proxy_http proxy_wstunnel headers ssl http2 expires filter brotli
+a2disconf other-vhosts-access-log
+{ crontab -l 2>/dev/null; echo "* * * * * chown -R www-data:www-data /var/www" ; } | crontab -
+cd /etc/apache2
+nano apache2.conf
+```
+
+Press <kbd>CTRL + W</kbd> to search for `<Directory /var/www/>`.<br>
+Go down a few lines to `AllowOverride None` and replace it with `AllowOverride All`. <br>
+
+```diff
+- AllowOverride None
++ AllowOverride All
+```
+
+Now add somewhere inside the file in a new line:
+```
+Protocols h2 h2c http/1.1
+SetOutputFilter BROTLI_COMPRESS
+SetEnvIfNoCase Request_URI \.(?:gif|jpe?g|png)$ no-brotli
+
+<IfModule mod_headers.c>
+    # Serve brotli compressed CSS files if they exist
+    # and the client accepts brotli.
+    RewriteCond "%{HTTP:Accept-encoding}" "br"
+    RewriteCond "%{REQUEST_FILENAME}\.br" "-s"
+    RewriteRule "^(.*)\.css"              "$1\.css\.br" [QSA]
+
+    # Serve brotli compressed JS files if they exist
+    # and the client accepts brotli.
+    RewriteCond "%{HTTP:Accept-encoding}" "br"
+    RewriteCond "%{REQUEST_FILENAME}\.br" "-s"
+    RewriteRule "^(.*)\.js"               "$1\.js\.br" [QSA]
+    
+    Header set Vary *
+
+    # Serve correct content types, and prevent double compression.
+    RewriteRule "\.css\.br$" "-" [T=text/css,E=no-brotli:1]
+    RewriteRule "\.js\.br$"  "-" [T=text/javascript,E=no-brotli:1]
+
+
+    <FilesMatch "(\.js\.br|\.css\.br)$">
+      # Serve correct encoding type.
+      Header append Content-Encoding br
+
+      # Force proxies to cache brotli &
+      # non-brotli css/js files separately.
+      Header append Vary Accept-Encoding
+    </FilesMatch>
+</IfModule>
+```
+
+Save your edit by pressing <kbd>CTRL + X</kbd>, then press <kbd>y</kbd>.<br>
+
+Now we generate a self signed certificate for not defined apache2 hosts
+```sh
+openssl req  -nodes -new -x509 -subj '/CN=localhost' -sha256 -subj '/CN=' -keyout /etc/ssl/privkey.pem -out /etc/ssl/fullchain.pem -days 365000
```

Comment:
The `-subj` parameter is given twice. This leads to the following error: 
```
root@server:/etc/apache2# openssl req  -nodes -new -x509 -subj '/CN=localhost' -sha256 -subj '/CN=' -keyout /etc/ssl/privkey.pem -out /etc/ssl/fullchain.pem -days 365000
Generating a RSA private key
.............................................................+++++
.....................................................................................................................................+++++
writing new private key to '/etc/ssl/privkey.pem'
-----
req: No value provided for Subject Attribute CN, skipped

```

I simply removed the second occurrence.

Reply:
fixed